lib/src/git_backend.rs

```diff
@@ -1352,13 +1352,20 @@ impl Backend for GitBackend {
             |change: gix::object::tree::diff::Change| -> BackendResult<Option<CopyRecord>> {
                 let gix::object::tree::diff::Change::Rewrite {
                     source_location,
+                    source_entry_mode,
                     source_id,
+                    entry_mode: dest_entry_mode,
                     location: dest_location,
                     ..
                 } = change
                 else {
                     return Ok(None);
                 };
+                // TODO: Renamed symlinks cannot be returned because CopyRecord
+                // expects `source_file: FileId`.
+                if !source_entry_mode.is_blob() || !dest_entry_mode.is_blob() {
+                    return Ok(None);
+                }
```

Comment:
I'm a bit surprised that we get tree entries from Gitoxide. I wonder if that's a bug or intentional. I can see it easily being a bug that happens due to comparing tree entries exactly as a first step.

Reply:
`gix` appears to emit directory rename entry even if new file is added to the directory. So I think it's intentional?